packages/@aws-cdk/aws-batch/lib/exposed-secret.ts

```diff
@@ -0,0 +1,92 @@
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as ssm from '@aws-cdk/aws-ssm';
+
+/**
+ * Properties for SecretsManagerSecret
+ */
+export interface SecretsManagerSecretProps {
+  /**
+   * The name of the secret.
+   */
+  readonly name: string;
+
+  /**
+   * A secret from secrets manager
+   */
+  readonly value: secretsmanager.ISecret;
+}
+
+/**
+ * Properties for ParametersStoreSecret
+ */
+export interface ParametersStoreSecretProps {
+  /**
+   * The name of the secret.
+   */
+  readonly name: string;
+
+  /**
+   * A parameter from parameters store
+   */
+  readonly value: ssm.IParameter;
+}
+
+/**
+ * Exposed secret for log configuration
+ */
+export abstract class ExposedSecret {
```

Comment:
I think these interfaces are not really necessary. Since it's just two properties, both of which are required, we can just put them as positional arguments in the `from` functions.

If optional properties are added in the future we could add an `Options` interface.